### DISCUSSION TITLE
NGINX not clearing next-auth cookies on sign out

### DISCUSSION THREAD

**Discussion by TheMythologist** [nextauthjs/next-auth#7431] [Help]
### Question 💬

Hi, my use case is that I am hosting a Next.js application with next-auth to serve as an SSO solution, which is routed via NGINX's `auth_request` directive. The log-out functionality doesn't work, i.e. I do get redirected back to the login page of my other subdomain, but the client-side cookie is still set.

When logging out directly via the application directly (i.e. through the application port directly) without routing through the NGINX reverse proxy, the following HTTP header is sent:
```http
set-cookie: next-auth.session-token=; Max-Age=0; Path=/; HttpOnly; SameSite=Lax
```

However, when I log out through NGINX itself, this HTTP header is not sent.

This is my NGINX configuration:
```nginx
server {
        location / {
                auth_request /auth; 
                auth_request_set $new_cookie $sent_http_set_cookie;
                add_header Set-Cookie $new_cookie;
                ...
                # Successful login, redirect to application
                proxy_pass http://127.0.0.1:8100;
        }

        location /auth {
                internal;
                proxy_pass https://sso.sol/api/auth/check;
                proxy_pass_request_body off;
                proxy_set_header Content-Length "";
                proxy_set_header X-Original-URI $request_uri;
        }
        error_page 401 = @error401;

        # If the user is not logged in, redirect them to SSO auth
        location @error401 {
                return 302 https:/sso.sol/auth/sign-in/?callbackUrl=https://$host$request_uri;
        }
}
```

Do let me know if there is any other code or configuration that I should provide, thanks.

### Contributing 🙌🏽

Yes, I am willing to help answer this question in a PR

**Comment by TheMythologist** [comment]
Hi, I did a closer inspection, and it turns out that the signout request issued by `signout` function is not being sent to the `NEXTAUTH_URL` endpoint. Is this intended?

**Comment by MarinBarac** [comment]
Did you manage to solve this problem?
